src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -3065,6 +3064,23 @@ public void report(DiagnosticPosition _unused, JCDiagnostic details) {
             };
         }
 
+        void checkNewInnerClass(DiagnosticPosition pos, Env<AttrContext> env, Type type) {
+            boolean isLocal = type.tsym.owner.kind == MTH;
+            if ((type.tsym.flags() & (INTERFACE | ENUM | RECORD)) != 0 ||
+                    (!isLocal && !type.tsym.isInner())) {
+                // nothing to check
+                return;
+            }
+            Symbol res = isLocal ?
+                    rs.findLocalClassOwner(env, type.tsym) :
+                    rs.findSelfContaining(pos, env, type.getEnclosingType().tsym, names._this);
+            if (res.exists()) {
+                rs.accessBase(res, pos, env.enclClass.sym.type, names._this, true);
```

Comment:
I wonder if the call to accesBase should be part of Resolve::findLocalClassOwner and Resolve::findSelfContaining? in other case clients needs to know a bit more about Resolve's internals. Although there are other places in Attr where we invoke Resolve:accessBase, so probably in the fence with this one